Food/Supporting/Network.swift

```
@@ -0,0 +1,26 @@
+//
+//  Network.swift
+//  Food
+//
+//  Created by Hanzheng Li on 4/22/21.
+//
+
+import Foundation
+import Apollo
+import Combine
+
+class Network {
+    static let shared = ApolloClient(url: URL(string: "https://graphql.anilist.co/")!)
+    
+    static func getMedia(for id: Int, _ completion: @escaping (Media) -> Void) {
+        Network.shared.fetch(query: DetailQuery(id: id)) { result in
+            switch result {
+            case .success(let result):
+                guard let media = result.data?.media else { return }
```

I think it would make more sense to raise an error in this case rather than fail silently.

---

The fetch() function takes in a closure that does not have a "throws" tag on it, so I actually cannot throw an error from within this set of curly braces. Any suggestions on how to deal with this?

---

Here's one option: `completion` should have a [`Result<Media, Error>`](https://developer.apple.com/documentation/swift/result).